feat: triage with AI (#16691)
stackrox/stackrox: .github/triage.md

Reviewer: Right, but unlike this one, the other ones are:
1. linked to from top-level README.md, and
2. related to PR reviews on GH

Personally I'd call it `/CI_TRIAGE.md` but up to you.

# StackRox CI Triage Setup Guide

This guide explains how to set up Claude Code for automated StackRox CI failure investigation and triage.

## Prerequisites

To enable automated CI triage capabilities, you need to configure the following MCP (Model Context Protocol) servers in Claude Code:

- [mcp-atlassian](https://github.com/sooperset/mcp-atlassian)
- [Github MCP](https://github.com/github/github-mcp-server)
- [Prow MCP](https://github.com/redhat-community-ai-tools/prowject)
Example MCPs configuration template:

- Install gopls https://go.dev/gopls/#installation
- Generate `JIRA_PERSONAL_TOKEN` here https://issues.redhat.com/secure/ViewProfile.jspa?selectedTab=com.atlassian.pats.pats-plugin:jira-user-personal-access-tokens
- Generate `CONFLUENCE_PERSONAL_TOKEN`  here https://spaces.redhat.com/plugins/personalaccesstokens/usertokens.action

> - Generate `GITHUB_TOKEN` folowing this instructions https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

Reviewer: ```suggestion
- Generate `GITHUB_TOKEN` folowing these instructions https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
```

> - Generate `GITHUB_TOKEN` folowing this instructions https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens

Reviewer: ```suggestion
- Generate `GITHUB_TOKEN` folowing this instructions https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
- Export the above as environment variables before launching `claude`
```

> - Remember to pin image vesrions with SHA

Reviewer: ```suggestion
- Remember to pin image version using a content digest
```

> - Configure Claude following this instructions https://docs.claude.com/en/docs/claude-code/mcp or Cursor https://cursor.com/docs/context/mcp (note: cursor have a different syntax for env resolution)

Reviewer: ```suggestion
- Follow instructions to [configure Claude code](https://docs.claude.com/en/docs/claude-code/mcp) or [Cursor](https://cursor.com/docs/context/mcp) (note: cursor have a different syntax for env resolution)
```

        "-e",
        "JIRA_URL",
        "-e",
        "JIRA_PERSONAL_TOKEN",
        "ghcr.io/sooperset/mcp-atlassian:<Use desired version SHA>"
      ],
      "env": {
        "CONFLUENCE_URL": "https://spaces.redhat.com/",
        "CONFLUENCE_PERSONAL_TOKEN": "${CONFLUENCE_PERSONAL_TOKEN}",
        "JIRA_URL": "https://issues.redhat.com",
        "JIRA_PERSONAL_TOKEN": "${JIRA_PERSONAL_TOKEN}"
      }
    },
    "github": {
      "type": "http",
      "url": "https://api.githubcopilot.com/mcp/",
      "headers": {
        "Authorization": "Bearer ${GITHUB_TOKEN}"
      }
    },
    "gopls-mcp": {
      "command": "gopls",
      "args": ["mcp"]
    }
  }
}
```

## Agent Configuration

The `stackrox-ci-failure-investigator` agent is automatically configured in `.claude/agents/` and provides:

- **Automated JIRA issue analysis** - Fetches ROX-XXXXX issues and analyzes comments for build IDs
- **Prow build log retrieval** - Downloads CI artifacts and logs for failure analysis
- **Root cause investigation** - Correlates test failures with service logs and source code
- **Team assignment** - Routes issues to appropriate teams based on failure patterns

> ## Usage

Reviewer: ```suggestion
## Usage

### Confirm MCP servers are functional

Run: `/mcp`.

In case of problems, try `/doctor`.

```

# StackRox CI Triage Setup Guide

This guide explains how to set up Claude Code for automated StackRox CI failure investigation and triage.

## Prerequisites

To enable automated CI triage capabilities, you need to configure the following MCP (Model Context Protocol) servers in Claude Code:

- [mcp-atlassian](https://github.com/sooperset/mcp-atlassian)
- [Github MCP](https://github.com/github/github-mcp-server)
- [Prow MCP](https://github.com/redhat-community-ai-tools/prowject)

> 

Author: ```suggestion
Example MCPs configuration template:

- Generate `JIRA_PERSONAL_TOKEN` here https://issues.redhat.com/secure/ViewProfile.jspa?selectedTab=com.atlassian.pats.pats-plugin:jira-user-personal-access-tokens
- Generate `CONFLUENCE_PERSONAL_TOKEN`  here https://spaces.redhat.com/plugins/personalaccesstokens/usertokens.action
- Generate `GITHUB_TOKEN` folowing this instructions https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens
- Remeber to pin image vesrions with SHA

```json
"mcpServers": {
    "mcp-atlassian": {
      "command": "docker",
      "args": [
        "run",
        "-i",
        "--rm",
        "-e",
        "CONFLUENCE_URL",
        "-e",
        "CONFLUENCE_PERSONAL_TOKEN",
        "-e",
        "JIRA_URL",
        "-e",
        "JIRA_PERSONAL_TOKEN",
        "ghcr.io/sooperset/mcp-atlassian:<Use desired version SHA>"
      ],
      "env": {
        "CONFLUENCE_URL": "https://spaces.redhat.com/",
        "CONFLUENCE_PERSONAL_TOKEN": "${CONFLUENCE_PERSONAL_TOKEN}",
        "JIRA_URL": "https://issues.redhat.com",
        "JIRA_PERSONAL_TOKEN": "${JIRA_PERSONAL_TOKEN}"
      }
    },
    "github": {
      "type": "http",
      "url": "https://api.githubcopilot.com/mcp/",
      "headers": {
        "Authorization": "Bearer ${GITHUB_TOKEN}"
      }
    },
    "gopls-mcp": {
      "command": "gopls",
      "args": ["mcp"]
    }
  }
```
